src/statmodels.jl

```diff
@@ -446,21 +446,36 @@ struct ConvergenceException{T<:Real} <: Exception
     iters::Int
     lastchange::T
     tol::T
-    function ConvergenceException{T}(iters, lastchange::T, tol::T) where T<:Real
+    msg::String
+    function ConvergenceException{T}(
+        iters, lastchange::T, tol::T, msg::String
+    ) where T<:Real
         if tol > lastchange
             throw(ArgumentError("Change must be greater than tol."))
         else
-            new(iters, lastchange, tol)
+            new(iters, lastchange, tol, msg)
         end
     end
 end
 
-ConvergenceException(iters, lastchange::T=NaN, tol::T=NaN) where {T<:Real} =
-    ConvergenceException{T}(iters, lastchange, tol)
+ConvergenceException(
+    iters, lastchange::T=NaN, tol::T=NaN, msg::AbstractString = ""
+) where {T<:Real} =
+    ConvergenceException{T}(iters, lastchange, tol, string(msg))
 
 function Base.showerror(io::IO, ce::ConvergenceException)
-    print(io, "failure to converge after $(ce.iters) iterations.")
+    print(io, "failure to converge after ", ce.iters, " iterations.")
```

Comment:
Why change this?

Reply:
It makes an intermediate string  where it's not necessary to do so. It's mentioned in Julia's [performance tips](https://docs.julialang.org/en/v1/manual/performance-tips/index.html#Avoid-string-interpolation-for-I/O-1).

Reply:
Yes but we don't care about performance here, it's an error path.

Reply:
Reverted.